### PR TITLE
plexamp: 4.8.3 -> 4.10.1

### PR DESCRIPTION
**Pull request by Redhawk18**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -17079,6 +17079,14 @@
...
     name = "Tim Engler";
   };
+  redhawk = {
+    email = "redhawk76767676@gmail.com";
+    github = "Redhawk18";
+    githubId = 77415970;
+    name = "Redhawk";
+  };
+
+
```
Comment (tomodachi94) [inline maintainers/maintainer-list.nix:17089]: Remove the extra blank lines for consistency in the entire file.

**Review by tomodachi94 — CHANGES_REQUESTED** [review]
A pedantic thing. Also, consider using `--rebase` when pulling from master, to avoid merge commits.

**Comment by tomodachi94** [comment]
Replaces https://github.com/NixOS/nixpkgs/pull/313297

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (Redhawk18) [reply maintainers/maintainer-list.nix:17089]: I'm having trouble amending the commit.

Comment (Redhawk18) [reply maintainers/maintainer-list.nix:17089]:
```bash
─redhawk💾Mythra─ ~/code/nixpkgs 
 └$ git status
On branch master
Your branch and 'origin/master' have diverged,
and have 1 and 1 different commits each, respectively.
  (use "git pull" if you want to integrate the remote branch with yours)

nothing to commit, working tree clean

─redhawk💾Mythra─ ~/code/nixpkgs 
 └$ git pull

```
Git pull keeps hanging

Comment (Redhawk18) [reply maintainers/maintainer-list.nix:17089]: Oh I let it do whatever git was doing and now its pulling, I hope it's working

Comment (Redhawk18) [reply maintainers/maintainer-list.nix:17089]: Nope I created a git nightmare

**Review by NobbZ — CHANGES_REQUESTED** [review]
Also make the actual update the third commit. Title "plexamp: 4.8.3 -> 4.10.1"

No need to mention the changelog (part of the version bump) nor the update script (as it now happened in an extra commit).

File: pkgs/applications/audio/plexamp/update-plexamp.sh [hunk 2]
```diff
@@ -5,7 +5,7 @@
 set -Eeuxo pipefail
 
 cleanup() {
-    rm -rf "$TMPDIR"
+  rm -rf "$TMPDIR"
```
[the remark this answers is not in the document]
Comment (Redhawk18) [reply pkgs/applications/audio/plexamp/update-plexamp.sh:8]: This is part of my editor setup, it auto formats I don't have control over it.

File: pkgs/applications/audio/plexamp/update-plexamp.sh [hunk 3]
```diff
@@ -42,7 +42,7 @@ if diff "$DEFAULT_NIX" "$WORKING_NIX"; then
 fi
 
 # update sha hash (convenietly provided)
-sed -i "s@sha.* = .*;@sha512 = \"$SHA512\";@g" "$WORKING_NIX"
+sed -i "s@hash.* = .*;@hash = \"sha512-$SHA512\";@g" "$WORKING_NIX"
```
[the remark this answers is not in the document]
Comment (Redhawk18) [reply pkgs/applications/audio/plexamp/update-plexamp.sh:45]: so commits
1. add myself to the maintainer list
2. fix the update script
3. update the app

How would I uncommit these and split them into that many commits?

[the remark this answers is not in the document]
Comment (tomodachi94) [reply pkgs/applications/audio/plexamp/update-plexamp.sh:45]: See this SO answer for details: https://stackoverflow.com/questions/6217156/break-a-previous-commit-into-multiple-commits#6217314

**Pull request closed without merging**